### ISSUE TITLE
Suggestion: terse automatic Ruby switch warning

### ISSUE THREAD

**Issue by davidsantos-br** [rvm/rvm#4911]
It seems that the current behavior when rvm automatically switches Ruby versions is to do it silently if it's by .ruby-version or, if it's because of a Gemfile, display that three-line warning about how Heroku does it too and with instructions on how not to show that warning again.

I think .ruby-version switching the Ruby version silently is not great, because it requires that the user remembers that there is a .ruby-version in the directory, which version it is set to use, and that the switch is automatic. At least the first two things can easily be forgotten by someone who deals with many projects.

The Gemfile warning in my opinion goes the opposite way by being too verbose, while not providing the most useful information: which version of Ruby is now active. Sure, an `rvm list` is easy enough, but it is even easier not having to issue a command at all.

I think showing a message like `RVM: switched to ruby-2.4.9 (Gemfile)` or `RVM: switched to ruby-2.7.1 (.ruby-version)` or `RVM: switched to ruby-2.7.0 (default)` when leaving a directory containing a .ruby-version file or a Gemfile, perhaps also including the active gemset if pertinent, would be better. It keeps the user aware that the interpreter was switched, it informs the user which interpreter was switched to, and it tells the user why that happened, all in a single short line.

Instructions on how to disable the warning probably don't need to be displayed again after the first time, since one of the problems I see with the current Gemfile warning is how verbose it is.

**Comment by pkuczynski** [comment]
I think you right and we should display it more clearly:

```
Found '{PATH}/.ruby-version' with version <2.4.9>
Now using ruby 2.4.9
```

or

```
Found '{PATH}/Gemfile' with version <2.4.9>
Now using ruby 2.4.9
```

There should be no need to display the info on how to hide this information. It should be pretty obvious users can find this in the manual.

Would you be keen to create PR with this change?

**Comment by davidsantos-br** [comment]
Sorry for the delay. Looking into it now.

**Comment by davidsantos-br** [comment]
Sorry, I don't know bash nearly well enough to do it correctly. Sure, I could just replace the warning in `__rvmrc_warning_display_for_Gemfile()` with
```bash
rvm_warn "RVM: Switching to ruby '${rvm_ruby_string}${rvm_gemset_name:+@$rvm_gemset_name}' (specified in '$1')."
```
or something to that effect, but that would only work for the Gemfile (i.e. not for versions.conf or .ruby-version), it would be shown even if it's redundant (if the Gemfile specifies the same version that's already in use), and it wouldn't be shown when the user leaves a project folder to let them know that it was back to the default. I wouldn't know how to make those changes while being sure I'm not breaking something.

**Comment by joshuacronemeyer** [comment]
I really like the idea of this feature, and would be up for making a PR for this. I notice that this thread is getting pretty old. I'll start to re-familiarize myself with the RVM codebase (it has been many many years since I contributed) meanwhile @pkuczynski can u confirm I should be adding this feature to RVM 1.x? I was reading the CONTRIBUTING.md guide and it signals no new features in 1.x

**Comment by pkuczynski** [comment]
Yes, needs to be added to 1.x. The next version is more like an idea from @mpapis but nobody really works on it...
